packages/pay/src/utils.ts

```diff
@@ -129,48 +47,52 @@ export class Int {
     this.value = n
   }
 
+  static from<T extends Int>(n: T): T
   static from(n: Long): Int
+  static from(n: NonNegativeInteger): Int
   static from(n: number): Int | undefined
+  static from(n: bigint): Int | undefined
   static from(n: string): Int | undefined
-  static from(n: BigNumber): Int | undefined
-  static from(n: Long | number | string | BigNumber): Int | undefined {
-    if (typeof n === 'string') {
+  static from(n: Int | bigint | number | string): Int | undefined // Necessary for amounts passed during setup
+  static from<T extends Long | Int | bigint | number | string>(n: T): Int | undefined {
+    if (n instanceof Int) {
+      return new Int(n.value)
+    } else if (typeof n === 'bigint') {
+      return Int.fromBigint(n)
+    } else if (typeof n === 'string') {
       return Int.fromString(n)
     } else if (typeof n === 'number') {
       return Int.fromNumber(n)
-    } else if (BigNumber.isBigNumber(n)) {
-      return Int.fromBigNumber(n)
-    } else {
+    } else if (Long.isLong(n)) {
       return Int.fromLong(n)
     }
   }
 
+  private static fromBigint(n: bigint): Int | undefined {
+    if (n >= 0) {
+      return new Int(n)
+    }
+  }
+
   private static fromString(n: string): Int | undefined {
     try {
-      const big = BigInt(n)
-      if (big >= 0) {
-        return new Int(big)
-      }
+      return Int.fromBigint(BigInt(n))
       // eslint-disable-next-line no-empty
     } catch (_) {}
   }
 
-  private static fromNumber(n: number): Int | undefined {
-    if (Number.isInteger(n) && n >= 0) {
+  private static fromNumber(n: NonNegativeInteger): Int
+  private static fromNumber(n: number): Int | undefined
+  private static fromNumber<T extends number>(n: T): Int | undefined {
+    if (isNonNegativeInteger(n)) {
```

Comment:
why are these >0 checks being done in the Int class rather than in the PositiveInt class?

Reply:
`Int` is >= 0, `PositiveInt` is > 0